perma_web/perma/views/common.py

```diff
@@ -206,40 +208,77 @@ def contact(request):
+                if affiliations:
+                    affiliation_string = ', '.join(affiliations)
+        return affiliation_string
+
+    def handle_registrar_fields(form):
+        if getattr(request.user, 'is_organization_user', ''):
+            registrars = set(org.registrar for org in request.user.organizations.all())
+            if len(registrars) > 1:
+                form.fields['registrar'].choices = [(registrar.email, registrar.name) for registrar in registrars]
+            if len(registrars) == 1:
+                form.fields['registrar'].widget = widgets.HiddenInput()
+                email = registrars.pop().email
+                form.fields['registrar'].initial = email
+                form.fields['registrar'].choices = [(email,email)]
+        else:
+            del form.fields['registrar']
+        return form
+
+    def should_send_to_registrar():
+        return settings.CONTACT_REGISTRARS and \
+               getattr(request.user, 'is_organization_user', '')
 
     if request.method == 'POST':
-        form = ContactForm(request.POST)
+        form = handle_registrar_fields(ContactForm(request.POST))
         if form.is_valid():
-            # If our form is valid, let's generate and email to our contact folks
-
+            # Assemble info for email
             from_address = form.cleaned_data['email']
             subject = "[perma-contact] " + form.cleaned_data['subject']
-            referer = form.cleaned_data['referer']
-
-            affiliation_string = ''
-            if request.user.is_authenticated():
-                if request.user.registrar:
-                    affiliation_string = "{} (Registrar)".format(request.user.registrar.name)
-                else:
-                    affiliations = ["{} ({})".format(org.name, org.registrar.name) for org in request.user.organizations.all().order_by('registrar')]
-                    if affiliations:
-                        affiliation_string = ', '.join(affiliations)
-
-            send_admin_email(
-                subject,
-                from_address,
-                request,
-                'email/admin/contact.txt',
-                {
-                    "message": form.cleaned_data['message'],
-                    "from_address": from_address,
-                    "referer": referer,
-                    "affiliation_string": affiliation_string
-                },
-                referer
-            )
-            # redirect to a new URL:
-            return HttpResponseRedirect(reverse('contact_thanks'))
+            context = {
+                "message": form.cleaned_data['message'],
+                "from_address": from_address,
+                "referer": form.cleaned_data['referer'],
+                "affiliation_string": affiliation_string()
+            }
+            if should_send_to_registrar():
+                # Send to all active registar users for registrar and cc Perma
+                reg_email = form.cleaned_data['registrar']
```

Comment:
So this is actually taken care of already. By using a ChoiceField, and by adding only the user's own registrars as field.choices, we ensure that form validation will fail if anything but one of the user's own registrars is posted. I have a test for this: test_contact_org_user_submit_invalid